Fix tests and revert additional sorting of converters

mvdbeek opened the pull request:
49dd02f should be the fix for the failing tests, we need to keep the headers.

I have reverted the manual re-ordering of the converters and added a test that demonstrates that when converting input for `format="bam"` the `sam_to_bam` converter will be used instead of the higher priority `sam_to_bam_native` (which isn't compatible with `bam` input).

mvdbeek committed:
Keep header when converting sam to bam

Otherwise the converted datasets are not valid BAM files.

mvdbeek committed:
Revert "fix spelling"

This reverts commit b139726ed380192ecf03f10e91220a0488b573f3.

mvdbeek committed:
Revert "implement priority converters"

This reverts commit 8928b34506c1ba1f4856fad82dc42aef0f7c68f6.

mvdbeek committed:
Move up sam_to_bam_native converter

That should allow implicit conversion from sam to bam_native for tools that
support bam_native.

mvdbeek committed:
Revert "resort the converters so that the accepted_formats are infront"

This reverts commit 446643159540f8db67dfd0752560b387b63fc4b8.

mvdbeek committed:
Enhance tests for sam_to_bam conversion

This demonstrates that the `sam_to_bam` converter will be used for an input
with `format="bam"`, and not `sam_to_bam_native`.

bgruening commented:
@mvdbeek thanks, but this will not work. Try with these inputs:
```
    <inputs>
        <param name="input1" type="data" format="sam" label="SAM file" optional="true"/>
        <param name="input2" type="data" format="bam_native" label="Unsorted BAM file" optional="true"/>
        <param name="input3" type="data" format="bam" label="Sorted BAM file" optional="true"/>
    </inputs>
```

With this PR you don't see the `(as bam_native)` and you don't get an unsorted bam_native. If you have `format="bam"` `sam_to_bam` should be always used. The same for `bam_native`, but this is not the case without the changes for me.

EDIT: forgot, try with the `sam_with_header.sam` file from the test-data folder.

mvdbeek commented:
You can run this as a functional test:
```
source .venv/bin/activate && GALAXY_TEST_TOOL_CONF=config/tool_conf.xml.sample,test/functional/tools/samples_tool_conf.xml python scripts/functional_tests.py  test/functional/test_toolbox.py:TestForTool_sam_to_bam_native_conversion
```

This is the test output, it is doing the right thing (no converters in the first test, `sam_to_bam_native` in the second test and `sam_to_bam` in the third test).
```
galaxy.tools DEBUG 2018-01-13 14:24:33,922 Validated and populated state for tool request (99.006 ms)
galaxy.tools.actions.upload DEBUG 2018-01-13 14:24:34,051 Persisted uploads (36.428 ms)
galaxy.tools.actions.upload DEBUG 2018-01-13 14:24:34,234 Checked and cleaned uploads (182.426 ms)
galaxy.tools.actions.upload_common INFO 2018-01-13 14:24:34,295 tool upload1 created job id 1
galaxy.tools.actions.upload DEBUG 2018-01-13 14:24:34,387 Created upload job (152.842 ms)
galaxy.tools.execute DEBUG 2018-01-13 14:24:34,387 Tool [upload1] created job [1] (372.533 ms)
galaxy.tools.execute DEBUG 2018-01-13 14:24:34,412 Executed 1 job(s) for tool upload1 request: (480.437 ms)
galaxy.jobs DEBUG 2018-01-13 14:24:35,178 (1) Working directory for job is: /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/1
galaxy.jobs.handler DEBUG 2018-01-13 14:24:35,294 (1) Dispatching to local runner
galaxy.jobs DEBUG 2018-01-13 14:24:35,477 (1) Persisting job destination (destination id: local)
galaxy.jobs.runners DEBUG 2018-01-13 14:24:35,554 Job [1] queued (260.014 ms)
galaxy.jobs.handler INFO 2018-01-13 14:24:35,867 (1) Job dispatched
galaxy.jobs.command_factory INFO 2018-01-13 14:24:37,941 Built script [/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/1/tool_script.sh] for tool command [python /Users/mvandenb/src/galaxy/test/functional/tools/upload.py /Users/mvandenb/src/galaxy /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/1/registry.xml /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpGr1b37 1:/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/1/dataset_1_files:/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/files/000/dataset_1.dat]
galaxy.jobs.runners DEBUG 2018-01-13 14:24:39,111 (1) command is: rm -rf working; mkdir -p working; cd working; /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/1/tool_script.sh; return_code=$?; cd '/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/1'; 
[ "$GALAXY_VIRTUAL_ENV" = "None" ] && GALAXY_VIRTUAL_ENV="$_GALAXY_VIRTUAL_ENV"; _galaxy_setup_environment True
python "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/1/set_metadata_U_Qwco.py" "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/1/registry.xml" "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/1/working/galaxy.json" "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/1/metadata_in_HistoryDatasetAssociation_1_C4PU70,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/1/metadata_kwds_HistoryDatasetAssociation_1_Gl3FWB,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/1/metadata_out_HistoryDatasetAssociation_1_qDq7ww,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/1/metadata_results_HistoryDatasetAssociation_1_KAZOi2,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/files/000/dataset_1.dat,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/1/metadata_override_HistoryDatasetAssociation_1_Xw0kn0" 5242880; sh -c "exit $return_code"
galaxy.jobs.runners.local DEBUG 2018-01-13 14:24:39,142 (1) executing job script: /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/1/galaxy_1.sh
galaxy.jobs DEBUG 2018-01-13 14:24:39,164 (1) Persisting job destination (destination id: local)
galaxy.jobs.runners.local DEBUG 2018-01-13 14:24:42,506 execution finished: /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/1/galaxy_1.sh
galaxy.model.metadata DEBUG 2018-01-13 14:24:43,826 loading metadata from file for: HistoryDatasetAssociation 1
galaxy.tools DEBUG 2018-01-13 14:24:45,353 Validated and populated state for tool request (131.714 ms)
galaxy.jobs INFO 2018-01-13 14:24:45,453 Collecting metrics for Job 1
galaxy.util.object_wrapper WARNING 2018-01-13 14:24:45,588 Unable to create dynamic subclass for <type 'instance'>, None: metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases
galaxy.jobs DEBUG 2018-01-13 14:24:45,613 job 1 ended (finish() executed in (3107.052 ms))
galaxy.model.metadata DEBUG 2018-01-13 14:24:45,652 Cleaning up external metadata files
galaxy.tools.actions INFO 2018-01-13 14:24:46,006 Handled output named bam_native_output for tool sam_to_bam_native_conversion (418.025 ms)
galaxy.tools.actions INFO 2018-01-13 14:24:46,166 Handled output named bam_output for tool sam_to_bam_native_conversion (159.364 ms)
galaxy.tools.actions INFO 2018-01-13 14:24:46,189 Added output datasets to history (22.785 ms)
galaxy.tools.actions INFO 2018-01-13 14:24:46,299 Verified access to datasets for Job[unflushed,tool_id=sam_to_bam_native_conversion] (14.234 ms)
galaxy.tools.actions INFO 2018-01-13 14:24:46,301 Setup for job Job[unflushed,tool_id=sam_to_bam_native_conversion] complete, ready to flush (112.444 ms)
galaxy.tools.actions INFO 2018-01-13 14:24:46,351 Flushed transaction for job Job[id=2,tool_id=sam_to_bam_native_conversion] (49.315 ms)
galaxy.tools.execute DEBUG 2018-01-13 14:24:46,351 Tool [sam_to_bam_native_conversion] created job [2] (902.036 ms)
galaxy.tools.execute DEBUG 2018-01-13 14:24:46,376 Executed 1 job(s) for tool sam_to_bam_native_conversion request: (1020.785 ms)
galaxy.jobs DEBUG 2018-01-13 14:24:47,126 (2) Working directory for job is: /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/2
galaxy.jobs.handler DEBUG 2018-01-13 14:24:47,432 (2) Dispatching to local runner
galaxy.jobs DEBUG 2018-01-13 14:24:47,830 (2) Persisting job destination (destination id: local)
galaxy.jobs.runners DEBUG 2018-01-13 14:24:47,846 Job [2] queued (413.966 ms)
galaxy.jobs.handler INFO 2018-01-13 14:24:47,898 (2) Job dispatched
galaxy.jobs.command_factory INFO 2018-01-13 14:24:48,216 Built script [/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/2/tool_script.sh] for tool command [samtools view -b -h -@ ${GALAXY_SLOTS:-2} -o '/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/files/000/dataset_2.dat' '/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/files/000/dataset_1.dat']
galaxy.jobs.runners DEBUG 2018-01-13 14:24:49,393 (2) command is: rm -rf working; mkdir -p working; cd working; /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/2/tool_script.sh; return_code=$?; cd '/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/2'; 
[ "$GALAXY_VIRTUAL_ENV" = "None" ] && GALAXY_VIRTUAL_ENV="$_GALAXY_VIRTUAL_ENV"; _galaxy_setup_environment True
python "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/2/set_metadata_krpkLd.py" "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/2/registry.xml" "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/2/working/galaxy.json" "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/2/metadata_in_HistoryDatasetAssociation_2_vTdq0t,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/2/metadata_kwds_HistoryDatasetAssociation_2_26REKI,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/2/metadata_out_HistoryDatasetAssociation_2_OY2uOH,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/2/metadata_results_HistoryDatasetAssociation_2_DuyKkf,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/files/000/dataset_2.dat,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/2/metadata_override_HistoryDatasetAssociation_2_2sVRGn" "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/2/metadata_in_HistoryDatasetAssociation_3_Bjna8m,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/2/metadata_kwds_HistoryDatasetAssociation_3_3Tptt4,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/2/metadata_out_HistoryDatasetAssociation_3_xR7jxh,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/2/metadata_results_HistoryDatasetAssociation_3_5KKSGu,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/files/000/dataset_3.dat,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/2/metadata_override_HistoryDatasetAssociation_3_mrxitV" 5242880; sh -c "exit $return_code"
galaxy.jobs.runners.local DEBUG 2018-01-13 14:24:49,419 (2) executing job script: /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/2/galaxy_2.sh
galaxy.jobs DEBUG 2018-01-13 14:24:49,435 (2) Persisting job destination (destination id: local)
galaxy.jobs.runners.local DEBUG 2018-01-13 14:24:51,662 execution finished: /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/2/galaxy_2.sh
galaxy.model.metadata DEBUG 2018-01-13 14:24:52,188 loading metadata from file for: HistoryDatasetAssociation 2
galaxy.jobs INFO 2018-01-13 14:24:53,291 Collecting metrics for Job 2
galaxy.jobs DEBUG 2018-01-13 14:24:53,313 job 2 ended (finish() executed in (1650.495 ms))
galaxy.model.metadata DEBUG 2018-01-13 14:24:53,322 Cleaning up external metadata files
galaxy.model.metadata DEBUG 2018-01-13 14:24:53,381 Failed to cleanup MetadataTempFile temp files from /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/2/metadata_out_HistoryDatasetAssociation_3_xR7jxh: No JSON object could be decoded
.galaxy.web.framework.webapp INFO 2018-01-13 14:25:02,469 Session authenticated using Galaxy master api key
galaxy.web.framework.webapp INFO 2018-01-13 14:25:02,489 Session authenticated using Galaxy master api key
galaxy.tools DEBUG 2018-01-13 14:25:02,819 Validated and populated state for tool request (97.017 ms)
galaxy.tools.actions.upload DEBUG 2018-01-13 14:25:02,896 Persisted uploads (43.864 ms)
galaxy.tools.actions.upload DEBUG 2018-01-13 14:25:03,375 Checked and cleaned uploads (479.006 ms)
galaxy.tools.actions.upload_common INFO 2018-01-13 14:25:03,432 tool upload1 created job id 3
galaxy.tools.actions.upload DEBUG 2018-01-13 14:25:03,520 Created upload job (143.904 ms)
galaxy.tools.execute DEBUG 2018-01-13 14:25:03,520 Tool [upload1] created job [3] (667.565 ms)
galaxy.tools.execute DEBUG 2018-01-13 14:25:03,544 Executed 1 job(s) for tool upload1 request: (724.970 ms)
galaxy.jobs DEBUG 2018-01-13 14:25:04,574 (3) Working directory for job is: /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/3
galaxy.jobs.handler DEBUG 2018-01-13 14:25:04,670 (3) Dispatching to local runner
galaxy.jobs DEBUG 2018-01-13 14:25:05,028 (3) Persisting job destination (destination id: local)
galaxy.jobs.runners DEBUG 2018-01-13 14:25:05,070 Job [3] queued (400.004 ms)
galaxy.jobs.handler INFO 2018-01-13 14:25:05,184 (3) Job dispatched
galaxy.jobs.command_factory INFO 2018-01-13 14:25:07,668 Built script [/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/3/tool_script.sh] for tool command [python /Users/mvandenb/src/galaxy/test/functional/tools/upload.py /Users/mvandenb/src/galaxy /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/3/registry.xml /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmp9R5CmY 4:/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/3/dataset_4_files:/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/files/000/dataset_4.dat]
galaxy.jobs.runners DEBUG 2018-01-13 14:25:09,240 (3) command is: rm -rf working; mkdir -p working; cd working; /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/3/tool_script.sh; return_code=$?; cd '/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/3'; 
[ "$GALAXY_VIRTUAL_ENV" = "None" ] && GALAXY_VIRTUAL_ENV="$_GALAXY_VIRTUAL_ENV"; _galaxy_setup_environment True
python "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/3/set_metadata_jts16_.py" "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/3/registry.xml" "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/3/working/galaxy.json" "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/3/metadata_in_HistoryDatasetAssociation_4_AE1J9B,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/3/metadata_kwds_HistoryDatasetAssociation_4_PauO2C,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/3/metadata_out_HistoryDatasetAssociation_4_heAEDz,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/3/metadata_results_HistoryDatasetAssociation_4_EAHePB,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/files/000/dataset_4.dat,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/3/metadata_override_HistoryDatasetAssociation_4_KP8b_s" 5242880; sh -c "exit $return_code"
galaxy.jobs.runners.local DEBUG 2018-01-13 14:25:09,261 (3) executing job script: /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/3/galaxy_3.sh
galaxy.jobs DEBUG 2018-01-13 14:25:09,288 (3) Persisting job destination (destination id: local)
galaxy.jobs.runners.local DEBUG 2018-01-13 14:25:12,582 execution finished: /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/3/galaxy_3.sh
galaxy.model.metadata DEBUG 2018-01-13 14:25:13,335 loading metadata from file for: HistoryDatasetAssociation 4
galaxy.tools DEBUG 2018-01-13 14:25:14,675 Validated and populated state for tool request (111.292 ms)
galaxy.jobs INFO 2018-01-13 14:25:14,746 Collecting metrics for Job 3
galaxy.jobs DEBUG 2018-01-13 14:25:15,419 job 3 ended (finish() executed in (2836.233 ms))
galaxy.model.metadata DEBUG 2018-01-13 14:25:15,521 Cleaning up external metadata files
galaxy.tools.actions INFO 2018-01-13 14:25:15,769 Handled output named output for tool CONVERTER_sam_to_bam_native (32.194 ms)
galaxy.tools.actions INFO 2018-01-13 14:25:15,776 Added output datasets to history (6.595 ms)
galaxy.tools.actions INFO 2018-01-13 14:25:15,808 Verified access to datasets for Job[unflushed,tool_id=CONVERTER_sam_to_bam_native] (12.325 ms)
galaxy.tools.actions INFO 2018-01-13 14:25:15,809 Setup for job Job[unflushed,tool_id=CONVERTER_sam_to_bam_native] complete, ready to flush (32.950 ms)
galaxy.tools.actions INFO 2018-01-13 14:25:15,845 Flushed transaction for job Job[id=4,tool_id=CONVERTER_sam_to_bam_native] (35.365 ms)
galaxy.tools.actions INFO 2018-01-13 14:25:15,941 Handled output named bam_native_output for tool sam_to_bam_native_conversion (32.157 ms)
galaxy.tools.actions INFO 2018-01-13 14:25:15,978 Handled output named bam_output for tool sam_to_bam_native_conversion (36.351 ms)
galaxy.tools.actions INFO 2018-01-13 14:25:16,000 Added output datasets to history (21.273 ms)
galaxy.tools.actions INFO 2018-01-13 14:25:16,038 Verified access to datasets for Job[unflushed,tool_id=sam_to_bam_native_conversion] (16.173 ms)
galaxy.tools.actions INFO 2018-01-13 14:25:16,039 Setup for job Job[unflushed,tool_id=sam_to_bam_native_conversion] complete, ready to flush (39.750 ms)
galaxy.tools.actions INFO 2018-01-13 14:25:16,126 Flushed transaction for job Job[id=5,tool_id=sam_to_bam_native_conversion] (86.480 ms)
galaxy.tools.execute DEBUG 2018-01-13 14:25:16,127 Tool [sam_to_bam_native_conversion] created job [5] (1052.881 ms)
galaxy.tools.execute DEBUG 2018-01-13 14:25:16,146 Executed 1 job(s) for tool sam_to_bam_native_conversion request: (1471.378 ms)
galaxy.jobs DEBUG 2018-01-13 14:25:16,768 (4) Working directory for job is: /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/4
galaxy.jobs.handler DEBUG 2018-01-13 14:25:16,782 (4) Dispatching to local runner
galaxy.jobs DEBUG 2018-01-13 14:25:16,844 (4) Persisting job destination (destination id: local)
galaxy.jobs.runners DEBUG 2018-01-13 14:25:16,864 Job [4] queued (81.460 ms)
galaxy.jobs.handler INFO 2018-01-13 14:25:16,921 (4) Job dispatched
galaxy.jobs.command_factory INFO 2018-01-13 14:25:18,028 Built script [/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/4/tool_script.sh] for tool command [samtools view -b -h -@ ${GALAXY_SLOTS:-2} -o '/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/files/000/dataset_5.dat' '/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/files/000/dataset_4.dat']
galaxy.jobs.runners DEBUG 2018-01-13 14:25:19,531 (4) command is: rm -rf working; mkdir -p working; cd working; /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/4/tool_script.sh; return_code=$?; cd '/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/4'; 
[ "$GALAXY_VIRTUAL_ENV" = "None" ] && GALAXY_VIRTUAL_ENV="$_GALAXY_VIRTUAL_ENV"; _galaxy_setup_environment True
python "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/4/set_metadata_5E60ja.py" "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/4/registry.xml" "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/4/working/galaxy.json" "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/4/metadata_in_HistoryDatasetAssociation_5_r6pU1Z,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/4/metadata_kwds_HistoryDatasetAssociation_5_CsuWB3,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/4/metadata_out_HistoryDatasetAssociation_5_kVIutt,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/4/metadata_results_HistoryDatasetAssociation_5_uquiCR,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/files/000/dataset_5.dat,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/4/metadata_override_HistoryDatasetAssociation_5_tjuHtn" 5242880; sh -c "exit $return_code"
galaxy.jobs.runners.local DEBUG 2018-01-13 14:25:19,553 (4) executing job script: /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/4/galaxy_4.sh
galaxy.jobs DEBUG 2018-01-13 14:25:19,570 (4) Persisting job destination (destination id: local)
galaxy.jobs.runners.local DEBUG 2018-01-13 14:25:21,688 execution finished: /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/4/galaxy_4.sh
galaxy.model.metadata DEBUG 2018-01-13 14:25:21,986 loading metadata from file for: HistoryDatasetAssociation 5
galaxy.jobs INFO 2018-01-13 14:25:22,234 Collecting metrics for Job 4
galaxy.jobs DEBUG 2018-01-13 14:25:22,258 job 4 ended (finish() executed in (569.176 ms))
galaxy.model.metadata DEBUG 2018-01-13 14:25:22,298 Cleaning up external metadata files
galaxy.jobs DEBUG 2018-01-13 14:25:23,036 (5) Working directory for job is: /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/5
galaxy.jobs.handler DEBUG 2018-01-13 14:25:23,192 (5) Dispatching to local runner
galaxy.jobs DEBUG 2018-01-13 14:25:23,335 (5) Persisting job destination (destination id: local)
galaxy.jobs.runners DEBUG 2018-01-13 14:25:23,353 Job [5] queued (159.856 ms)
galaxy.jobs.handler INFO 2018-01-13 14:25:23,407 (5) Job dispatched
galaxy.jobs.command_factory INFO 2018-01-13 14:25:23,714 Built script [/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/5/tool_script.sh] for tool command [cp '/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/files/000/dataset_5.dat' '/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/files/000/dataset_6.dat']
galaxy.jobs.runners DEBUG 2018-01-13 14:25:24,262 (5) command is: rm -rf working; mkdir -p working; cd working; /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/5/tool_script.sh; return_code=$?; cd '/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/5'; 
[ "$GALAXY_VIRTUAL_ENV" = "None" ] && GALAXY_VIRTUAL_ENV="$_GALAXY_VIRTUAL_ENV"; _galaxy_setup_environment True
python "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/5/set_metadata_oEXg4T.py" "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/5/registry.xml" "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/5/working/galaxy.json" "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/5/metadata_in_HistoryDatasetAssociation_6_mDNXPU,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/5/metadata_kwds_HistoryDatasetAssociation_6_H19RcY,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/5/metadata_out_HistoryDatasetAssociation_6_IggqLW,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/5/metadata_results_HistoryDatasetAssociation_6_adNiJl,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/files/000/dataset_6.dat,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/5/metadata_override_HistoryDatasetAssociation_6_UlY_N9" "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/5/metadata_in_HistoryDatasetAssociation_7_khcsKd,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/5/metadata_kwds_HistoryDatasetAssociation_7_Udp_xF,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/5/metadata_out_HistoryDatasetAssociation_7_aTxhT7,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/5/metadata_results_HistoryDatasetAssociation_7_I0thch,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/files/000/dataset_7.dat,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/5/metadata_override_HistoryDatasetAssociation_7_q90sOG" 5242880; sh -c "exit $return_code"
galaxy.jobs.runners.local DEBUG 2018-01-13 14:25:24,283 (5) executing job script: /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/5/galaxy_5.sh
galaxy.jobs DEBUG 2018-01-13 14:25:24,302 (5) Persisting job destination (destination id: local)
galaxy.jobs.runners.local DEBUG 2018-01-13 14:25:26,570 execution finished: /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/5/galaxy_5.sh
galaxy.model.metadata DEBUG 2018-01-13 14:25:27,525 loading metadata from file for: HistoryDatasetAssociation 6
galaxy.jobs INFO 2018-01-13 14:25:27,810 Collecting metrics for Job 5
galaxy.jobs DEBUG 2018-01-13 14:25:27,834 job 5 ended (finish() executed in (1264.094 ms))
galaxy.model.metadata DEBUG 2018-01-13 14:25:27,844 Cleaning up external metadata files
galaxy.model.metadata DEBUG 2018-01-13 14:25:27,932 Failed to cleanup MetadataTempFile temp files from /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/5/metadata_out_HistoryDatasetAssociation_7_aTxhT7: No JSON object could be decoded
.galaxy.web.framework.webapp INFO 2018-01-13 14:25:40,140 Session authenticated using Galaxy master api key
galaxy.web.framework.webapp INFO 2018-01-13 14:25:40,163 Session authenticated using Galaxy master api key
galaxy.tools DEBUG 2018-01-13 14:25:40,477 Validated and populated state for tool request (86.381 ms)
galaxy.tools.actions.upload DEBUG 2018-01-13 14:25:40,603 Persisted uploads (40.684 ms)
galaxy.tools.actions.upload DEBUG 2018-01-13 14:25:40,813 Checked and cleaned uploads (209.379 ms)
galaxy.tools.actions.upload_common INFO 2018-01-13 14:25:40,876 tool upload1 created job id 6
galaxy.tools.actions.upload DEBUG 2018-01-13 14:25:41,242 Created upload job (428.552 ms)
galaxy.tools.execute DEBUG 2018-01-13 14:25:41,242 Tool [upload1] created job [6] (679.753 ms)
galaxy.tools.execute DEBUG 2018-01-13 14:25:41,263 Executed 1 job(s) for tool upload1 request: (785.294 ms)
galaxy.jobs DEBUG 2018-01-13 14:25:42,467 (6) Working directory for job is: /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/6
galaxy.jobs.handler DEBUG 2018-01-13 14:25:42,633 (6) Dispatching to local runner
galaxy.jobs DEBUG 2018-01-13 14:25:42,995 (6) Persisting job destination (destination id: local)
galaxy.jobs.runners DEBUG 2018-01-13 14:25:43,032 Job [6] queued (399.541 ms)
galaxy.jobs.handler INFO 2018-01-13 14:25:43,216 (6) Job dispatched
galaxy.jobs.command_factory INFO 2018-01-13 14:25:45,449 Built script [/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/6/tool_script.sh] for tool command [python /Users/mvandenb/src/galaxy/test/functional/tools/upload.py /Users/mvandenb/src/galaxy /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/6/registry.xml /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpzHmI0C 8:/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/6/dataset_8_files:/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/files/000/dataset_8.dat]
galaxy.jobs.runners DEBUG 2018-01-13 14:25:46,992 (6) command is: rm -rf working; mkdir -p working; cd working; /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/6/tool_script.sh; return_code=$?; cd '/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/6'; 
[ "$GALAXY_VIRTUAL_ENV" = "None" ] && GALAXY_VIRTUAL_ENV="$_GALAXY_VIRTUAL_ENV"; _galaxy_setup_environment True
python "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/6/set_metadata_9zE9Jd.py" "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/6/registry.xml" "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/6/working/galaxy.json" "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/6/metadata_in_HistoryDatasetAssociation_8_G8EP83,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/6/metadata_kwds_HistoryDatasetAssociation_8_7KIeN7,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/6/metadata_out_HistoryDatasetAssociation_8_j4yMFF,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/6/metadata_results_HistoryDatasetAssociation_8_5Goy9N,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/files/000/dataset_8.dat,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/6/metadata_override_HistoryDatasetAssociation_8_XJ0VfY" 5242880; sh -c "exit $return_code"
galaxy.jobs.runners.local DEBUG 2018-01-13 14:25:47,015 (6) executing job script: /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/6/galaxy_6.sh
galaxy.jobs DEBUG 2018-01-13 14:25:47,032 (6) Persisting job destination (destination id: local)
galaxy.jobs.runners.local DEBUG 2018-01-13 14:25:50,222 execution finished: /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/6/galaxy_6.sh
galaxy.model.metadata DEBUG 2018-01-13 14:25:51,504 loading metadata from file for: HistoryDatasetAssociation 8
galaxy.tools DEBUG 2018-01-13 14:25:53,530 Validated and populated state for tool request (276.909 ms)
galaxy.jobs INFO 2018-01-13 14:25:53,655 Collecting metrics for Job 6
galaxy.jobs DEBUG 2018-01-13 14:25:54,002 job 6 ended (finish() executed in (3779.675 ms))
galaxy.model.metadata DEBUG 2018-01-13 14:25:54,075 Cleaning up external metadata files
galaxy.tools.actions INFO 2018-01-13 14:25:54,680 Handled output named output for tool CONVERTER_sam_to_bam (423.722 ms)
galaxy.tools.actions INFO 2018-01-13 14:25:54,688 Added output datasets to history (6.894 ms)
galaxy.tools.actions INFO 2018-01-13 14:25:54,720 Verified access to datasets for Job[unflushed,tool_id=CONVERTER_sam_to_bam] (15.119 ms)
galaxy.tools.actions INFO 2018-01-13 14:25:54,722 Setup for job Job[unflushed,tool_id=CONVERTER_sam_to_bam] complete, ready to flush (33.971 ms)
galaxy.tools.actions INFO 2018-01-13 14:25:54,760 Flushed transaction for job Job[id=7,tool_id=CONVERTER_sam_to_bam] (37.640 ms)
galaxy.tools.actions INFO 2018-01-13 14:25:54,856 Handled output named bam_native_output for tool sam_to_bam_native_conversion (31.400 ms)
galaxy.tools.actions INFO 2018-01-13 14:25:54,890 Handled output named bam_output for tool sam_to_bam_native_conversion (34.239 ms)
galaxy.tools.actions INFO 2018-01-13 14:25:54,911 Added output datasets to history (20.679 ms)
galaxy.tools.actions INFO 2018-01-13 14:25:55,113 Verified access to datasets for Job[unflushed,tool_id=sam_to_bam_native_conversion] (100.451 ms)
galaxy.tools.actions INFO 2018-01-13 14:25:55,115 Setup for job Job[unflushed,tool_id=sam_to_bam_native_conversion] complete, ready to flush (203.309 ms)
galaxy.tools.actions INFO 2018-01-13 14:25:55,166 Flushed transaction for job Job[id=8,tool_id=sam_to_bam_native_conversion] (51.433 ms)
galaxy.tools.execute DEBUG 2018-01-13 14:25:55,167 Tool [sam_to_bam_native_conversion] created job [8] (1526.586 ms)
galaxy.tools.execute DEBUG 2018-01-13 14:25:55,188 Executed 1 job(s) for tool sam_to_bam_native_conversion request: (1658.156 ms)
galaxy.jobs DEBUG 2018-01-13 14:25:55,958 (7) Working directory for job is: /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/7
galaxy.jobs.handler DEBUG 2018-01-13 14:25:56,098 (7) Dispatching to local runner
galaxy.jobs DEBUG 2018-01-13 14:25:56,385 (7) Persisting job destination (destination id: local)
galaxy.jobs.runners DEBUG 2018-01-13 14:25:56,554 Job [7] queued (454.007 ms)
galaxy.jobs.handler INFO 2018-01-13 14:25:56,653 (7) Job dispatched
galaxy.jobs.command_factory INFO 2018-01-13 14:25:57,141 Built script [/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/7/tool_script.sh] for tool command [python '/Users/mvandenb/src/galaxy/lib/galaxy/datatypes/converters/sam_to_bam.py' '/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/files/000/dataset_8.dat' '/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/files/000/dataset_9.dat']
galaxy.jobs.runners DEBUG 2018-01-13 14:25:58,255 (7) command is: rm -rf working; mkdir -p working; cd working; /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/7/tool_script.sh; return_code=$?; cd '/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/7'; 
[ "$GALAXY_VIRTUAL_ENV" = "None" ] && GALAXY_VIRTUAL_ENV="$_GALAXY_VIRTUAL_ENV"; _galaxy_setup_environment True
python "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/7/set_metadata_SDv90n.py" "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/7/registry.xml" "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/7/working/galaxy.json" "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/7/metadata_in_HistoryDatasetAssociation_9_FPwbmX,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/7/metadata_kwds_HistoryDatasetAssociation_9_1ikaZ0,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/7/metadata_out_HistoryDatasetAssociation_9_wr6izA,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/7/metadata_results_HistoryDatasetAssociation_9_2Xa4rb,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/files/000/dataset_9.dat,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/7/metadata_override_HistoryDatasetAssociation_9_zkNlVm" 5242880; sh -c "exit $return_code"
galaxy.jobs.runners.local DEBUG 2018-01-13 14:25:58,281 (7) executing job script: /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/7/galaxy_7.sh
galaxy.jobs DEBUG 2018-01-13 14:25:58,298 (7) Persisting job destination (destination id: local)
galaxy.jobs.runners.local DEBUG 2018-01-13 14:26:00,501 execution finished: /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/7/galaxy_7.sh
galaxy.model.metadata DEBUG 2018-01-13 14:26:01,399 loading metadata from file for: HistoryDatasetAssociation 9
galaxy.jobs DEBUG 2018-01-13 14:26:02,436 (8) Working directory for job is: /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/8
galaxy.jobs INFO 2018-01-13 14:26:02,460 Collecting metrics for Job 7
galaxy.jobs.handler DEBUG 2018-01-13 14:26:02,530 (8) Dispatching to local runner
galaxy.jobs DEBUG 2018-01-13 14:26:02,569 job 7 ended (finish() executed in (2067.512 ms))
galaxy.model.metadata DEBUG 2018-01-13 14:26:02,637 Cleaning up external metadata files
galaxy.jobs DEBUG 2018-01-13 14:26:02,796 (8) Persisting job destination (destination id: local)
galaxy.jobs.runners DEBUG 2018-01-13 14:26:02,828 Job [8] queued (297.307 ms)
galaxy.jobs.handler INFO 2018-01-13 14:26:03,012 (8) Job dispatched
galaxy.jobs.command_factory INFO 2018-01-13 14:26:03,636 Built script [/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/8/tool_script.sh] for tool command [cp '/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/files/000/dataset_9.dat' '/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/files/000/dataset_11.dat']
galaxy.jobs.runners DEBUG 2018-01-13 14:26:04,151 (8) command is: rm -rf working; mkdir -p working; cd working; /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/8/tool_script.sh; return_code=$?; cd '/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/8'; 
[ "$GALAXY_VIRTUAL_ENV" = "None" ] && GALAXY_VIRTUAL_ENV="$_GALAXY_VIRTUAL_ENV"; _galaxy_setup_environment True
python "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/8/set_metadata_prTUCh.py" "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/8/registry.xml" "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/8/working/galaxy.json" "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/8/metadata_in_HistoryDatasetAssociation_10_hh3vH6,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/8/metadata_kwds_HistoryDatasetAssociation_10_FtbLn9,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/8/metadata_out_HistoryDatasetAssociation_10_i49XDb,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/8/metadata_results_HistoryDatasetAssociation_10_9YLTCl,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/files/000/dataset_10.dat,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/8/metadata_override_HistoryDatasetAssociation_10_BuDxng" "/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/8/metadata_in_HistoryDatasetAssociation_11_aB0aTT,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/8/metadata_kwds_HistoryDatasetAssociation_11_HRoFAc,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/8/metadata_out_HistoryDatasetAssociation_11_6_OcJY,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/8/metadata_results_HistoryDatasetAssociation_11_kNPA3r,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/files/000/dataset_11.dat,/var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/8/metadata_override_HistoryDatasetAssociation_11_MVD1eO" 5242880; sh -c "exit $return_code"
galaxy.jobs.runners.local DEBUG 2018-01-13 14:26:04,173 (8) executing job script: /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/8/galaxy_8.sh
galaxy.jobs DEBUG 2018-01-13 14:26:04,190 (8) Persisting job destination (destination id: local)
galaxy.jobs.runners.local DEBUG 2018-01-13 14:26:06,305 execution finished: /var/folders/df/6xqpqpcd7h73b6jpx9t6cwhw0000gn/T/tmpAKSuYu/tmppO_0dT/tmpgHLqzm/database/job_working_directory_BUhub9/000/8/galaxy_8.sh
galaxy.model.metadata DEBUG 2018-01-13 14:26:06,965 loading metadata from file for: HistoryDatasetAssociation 11
galaxy.jobs INFO 2018-01-13 14:26:07,268 Collecting metrics for Job 8
galaxy.jobs DEBUG 2018-01-13 14:26:07,290 job 8 ended (finish() executed in (985.097 ms))
galaxy.model.metadata DEBUG 2018-01-13 14:26:07,300 Cleaning up external metadata files
```

mvdbeek commented:
Or interactively:
<img width="911" alt="screen shot 2018-01-13 at 23 32 44" src="https://user-images.githubusercontent.com/6804901/34910654-24073b48-f8ba-11e7-9139-0e93c796ebec.png">
